cupy/core/_routines_indexing.pyx

```diff
@@ -152,6 +152,22 @@ cdef ndarray _ndarray_choose(ndarray self, choices, out, mode):
     return out
 
 
+cdef ndarray _ndarray_compress(ndarray self, condition, axis, out):
+    a = self
+
+    if numpy.isscalar(condition):
+        raise ValueError('condition must be a 1-d array')
+
+    if not isinstance(condition, ndarray):
+        condition = core.array(condition, dtype=int)
+        if condition.ndim != 1:
+            raise ValueError('condition must be a 1-d array')
+
+    res = _ndarray_nonzero(condition)
```

Comment:
Let's leave a comment to note that this line causes host-device synchronization.
```suggestion
    res = _ndarray_nonzero(condition)  # synchronize
```